src/execution_engine.rs

```diff
@@ -89,7 +91,7 @@ impl ExecutionEngine {
     ///
     /// assert_eq!(result, 128.);
     /// ```
-    pub fn add_global_mapping(&self, value: &AnyValue, addr: usize) {
+    pub fn add_global_mapping(&mut self, value: &AnyValue, addr: usize) {
```

Comment:
I wonder if adding mut will just make the API more difficult to use, it doesn't actually add any guarantees since we're pretty much just working with interior mutability anyway...

Reply:
You raise a good point. Initially it felt quite strange to be mutating an execution engine without declaring it as `mut`, but it's probably better to remove the `&mut self` to stay consistent with the rest of the library.

Reply:
Yeah, I think we should remove it